qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -148,6 +151,129 @@ def model_names(self) -> List[str]:
         """Return model names."""
         return [getattr(m, "_name", f"model-{i}") for i, m in enumerate(self._models)]
 
+    def set_options(self, **fields):
+        """Set the analysis options for :meth:`run` method.
+
+        Args:
+            fields: The fields to update the options
+
+        Raises:
+            KeyError: When removed option ``curve_fitter`` is set.
+        """
+        if fields.get("plot_residuals") and not self.options.get("plot_residuals"):
+            # checking there are no subplots for the figure to prevent collision in subplot indices.
+            if self.plotter.options.get("subplots") != (1, 1):
+                warnings.warn(
+                    "Residuals plotting is currently supported for analysis with 1 subplot.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+                fields["plot_residuals"] = False
+            else:
+                self._add_residuals_plot_config()
+        if not fields.get("plot_residuals", True) and self.options.get("plot_residuals"):
+            self._remove_residuals_plot_config()
+
+        super().set_options(**fields)
+
+    def _add_residuals_plot_config(self):
+        """Configure plotter options for residuals plot."""
+        # check we have model to fit into
+        if self.models:
+            # Cache figure options.
+            self._plot_config_cache["figure_options"] = {}
+            self._plot_config_cache["figure_options"]["ylabel"] = self.plotter.figure_options.get(
+                "ylabel"
+            )
+            self._plot_config_cache["figure_options"][
+                "series_params"
+            ] = self.plotter.figure_options["series_params"]
+            self._plot_config_cache["figure_options"]["sharey"] = self.plotter.figure_options[
+                "sharey"
+            ]
+
+            self.plotter.set_figure_options(
+                ylabel=[
+                    self.plotter.figure_options.get("ylabel", ""),
+                    "Residuals",
+                ],
+            )
+            model_names = self.model_names()
+            for model_name in model_names:
+                self.plotter.set_figure_options(
+                    sharey=False,
+                    series_params={
+                        **self.plotter.figure_options["series_params"],
+                        **{
+                            model_name: {
+                                **{
+                                    "canvas": 0,
+                                },
+                                **self.plotter.figure_options["series_params"].get(model_name, {}),
+                            },
```

Comment:
It looks like you could replace the previous lines with
```python
for model in model_names:
    series_params = self.plotter.figure_options["series_params"]
    if series_params[model_name]:
        series_params[model_name]["canvas"] = 0
    else:
        series_params[model_name] = {"canvas": 0}
    self.plotter.set_figure_options(sharey=False, series_params=series_params)
```

Reply:
Sometimes, `series_params[model_name]` isn't defined (usually if there is only one curve in the plot). In addition, `series_params` is responsible on the options of the data in the figure (like colors).

Example 1: in `T2Hahn` experiment there is no `series_params[model_name]` while in `RamseyXY` there are 2 `model_name`.

Example 2: For `QuantumVolume` we can see that `series_params` is defined (in [qv_analysis.py](https://github.com/Qiskit-Extensions/qiskit-experiments/blob/main/qiskit_experiments/library/quantum_volume/qv_analysis.py#L83-L88) but without `model_name`:

```python
...
options.series_params = {
            "hop": {"color": "gray", "symbol": "."},
            "threshold": {"color": "black", "linestyle": "dashed", "linewidth": 1},
            "hop_cumulative": {"color": "r"},
            "hop_twosigma": {"color": "lightgray"},
        }
...
```

So to sum it up, we would like to keep `series_params`  that the user defined unchanged so we use the union of the `series_params` field and the option defined for each `model_name`.